notebooks/tests/test2_tellseq_C_equal_volume_pooling.py

```diff
@@ -0,0 +1,159 @@
+DATA_EXTS = {".txt", ".csv", ".tsv"}
+
+
+class TestTellseqC_ShadowWorkspace(unittest.TestCase):
+    def setUp(self):
+        # project root (parent of this test file’s folder)
+        self.notebooks_dir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+        self.repo_test_output_dir = os.path.join(self.notebooks_dir, "test_output")
+
+        # sanity: required inputs exist in repo
+        must_inputs = [
+            os.path.join(self.repo_test_output_dir, "QC", "Tellseq_plate_df_B.txt"),
+            os.path.join(self.repo_test_output_dir, "QC", "Tellseq_expt_info.yml"),
+        ]
+        for fp in must_inputs:
+            if not os.path.isfile(fp):
+                raise FileNotFoundError(f"[SETUP] Required input not found in repo: {fp}")
+
+    def _find_expected_path(self, filename: str) -> str:
+        """Find golden file by filename under test_output/<subdir>/*."""
+        for sub in EXPECTED_SUBDIRS:
+            candidate = os.path.join(self.repo_test_output_dir, sub, filename)
+            if os.path.exists(candidate):
+                return candidate
+        raise AssertionError(
+            f"[COMPARE] Expected golden file not found for '{filename}'. "
+            f"Searched under: {', '.join(EXPECTED_SUBDIRS)}"
+        )
+
+    def _assert_text_files_equal(self, produced_fp: Path, expected_fp: Path):
+        """Strict line-by-line text comparison with helpful diffs."""
+        with open(produced_fp, "r") as f_out, open(expected_fp, "r") as f_exp:
+            out_lines = f_out.readlines()
+            exp_lines = f_exp.readlines()
+
+        self.assertEqual(
+            len(out_lines), len(exp_lines),
+            msg=(f"[DIFF] Line count differs for '{produced_fp.name}': "
+                 f"produced={len(out_lines)} vs expected={len(exp_lines)}")
+        )
+
+        for i, (o, e) in enumerate(zip(out_lines, exp_lines), start=1):
+            if o != e:
+                self.fail(
+                    f"[DIFF] Mismatch in '{produced_fp.name}' at line {i}.\n"
+                    f"Produced: {o.rstrip()}\n"
+                    f"Expected: {e.rstrip()}\n"
+                )
+
+    def test_equal_volume_pooling_outputs_shadow(self):
+        """
+        Keep Notebook C's hardcoded './test_output/...' as-is by creating that
+        structure under a temp dir and running with cwd=tmp. Then compare tmp outputs
+        vs repo goldens by filename.
+        """
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            tmp_root = Path(tmp_dir)            # shadow root = './'
+            tmp_out  = tmp_root / "test_output" # ./test_output
+            tmp_qc   = tmp_out / "QC"
+            tmp_idx  = tmp_out / "Indices"
+            tmp_ss   = tmp_out / "SampleSheets"
+
+            # create expected dirs under tmp
+            for d in (tmp_qc, tmp_idx, tmp_ss):
+                d.mkdir(parents=True, exist_ok=True)
+
+            # copy required inputs from repo -> tmp
```

Comment:
There is no need to do this--you can use the inputs straight from their current location!